src/main.cpp

```diff
@@ -575,7 +574,178 @@ void affect_nearby_friends_with_alarm(struct Thing *traptng)
 
 long apply_wallhug_force_to_boulder(struct Thing *thing)
 {
-  return _DK_apply_wallhug_force_to_boulder(thing);
+  // return _DK_apply_wallhug_force_to_boulder(thing);
+  int i = 4 * thing->move_angle_xy;
+  unsigned short angle;
+  long collide;
+  unsigned short new_angle;
+  struct Coord3d pos2;
+  struct Coord3d pos;
+  struct ShotStats *shotst = &shot_stats[thing->model];
+  unsigned short speed = (unsigned short)shotst->speed;
+  *(unsigned short *)&pos.x.stl = *(unsigned short *)&thing->mappos.x.stl
+                           + ((speed * *(int *)((char *)lbSinTable + i)) >> 16);
+  int n = speed * *(int *)((char *)lbCosTable + i);
+  *(unsigned short *)&pos.y.stl = *(unsigned short *)&thing->mappos.y.stl + (-(n >> 8) >> 8);
+  *(unsigned short *)&pos.z.stl = *(unsigned short *)&thing->mappos.z.stl;
+  unsigned long seed = 9377 * game.action_rand_seed + 9439;
+  game.action_rand_seed = LB_RANDOM(4294967296, &seed);
```

Comment:
In KeeperFX we have random functions for this right?

Reply:
Both 581 and 592 should be turned into some of random macro. i.e. ACTION_RANDOM